lib/course_planner/notifier.ex

```diff
@@ -32,7 +16,18 @@ defmodule CoursePlanner.Notifier do
     GenServer.cast(__MODULE__, {:send_email, notification})
   end
 
-  @spec handle_cast({atom(), Notification.t}, any()) :: {:noreply, any()}
+  @spec notify_later(Notification.t) :: GenServer.cast
+  def notify_later(%Notification{} = notification) do
+    GenServer.cast(__MODULE__, {:save_email, notification})
+  end
+
+  @spec notify_all(User.t) :: GenServer.cast
+  def notify_all(%User{notifications: []} = _user), do: :nothing
+  def notify_all(%User{} = user) do
+    GenServer.cast(__MODULE__, {:notify_all, user})
+  end
+
+  @spec handle_cast({atom(), Notification.t} | atom(), any()) :: {:noreply, any()}
   def handle_cast({:send_email, notification}, state) do
```

Comment:
My idea was to keep it in case we add any type of individual notification or even if we add an option to notify immediately instead of in a summary.